src/main/java/chess/dao/MySqlChessGameDao.java

```diff
@@ -0,0 +1,138 @@
+package chess.dao;
+
+import chess.database.DatabaseConnector;
+import chess.domain.Board;
+import chess.domain.ChessGame;
+import chess.domain.Turn;
+import chess.domain.piece.Color;
+import chess.domain.piece.Piece;
+import chess.domain.piece.PieceFactory;
+import chess.domain.piece.PieceType;
+import chess.domain.square.File;
+import chess.domain.square.Rank;
+import chess.domain.square.Square;
+import chess.dto.ChessGameDto;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MySqlChessGameDao implements ChessGameDao {
+
+    private static final int STRING_FIRST_INDEX = 0;
+
+    private final DatabaseConnector databaseConnector;
+
+    public MySqlChessGameDao(final DatabaseConnector databaseConnector) {
+        this.databaseConnector = databaseConnector;
+    }
+
+    @Override
+    public int save(final ChessGameDto chessGameDto) {
+        try (final Connection connection = databaseConnector.getConnection()) {
+            int chessGameId = saveChessGame(chessGameDto, connection);
+            savePiece(chessGameDto.getBoard(), connection, chessGameId);
+            return chessGameId;
+        } catch (SQLException | IllegalStateException e) {
+            e.printStackTrace();
+            throw new RuntimeException("정상 저장되지 않았습니다.");
+        }
+    }
+
+    private int saveChessGame(final ChessGameDto chessGameDto, final Connection connection) {
+        final String query = "INSERT INTO chess_game(turn) VALUES (?)";
+        try (final PreparedStatement ps = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);) {
```

Comment:
```suggestion
        try (final PreparedStatement ps = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
```

세미콜론은 없어도 될 것 같아요 ㅎㅎ